src/kubernetes_cluster/spec/kubernetes_api/state_machine.rs

```diff
@@ -87,12 +89,14 @@ pub open spec fn validate_create_request(req: CreateRequest, s: KubernetesAPISta
     } else if req.obj.metadata.namespace.is_Some() && req.namespace != req.obj.metadata.namespace.get_Some_0() {
         // Creation fails because the namespace of the provided object does not match the namespace sent on the request
         Option::Some(APIError::BadRequest)
-    } else if !object_has_well_formed_spec(req.obj) {
+    } else if !Self::object_has_well_formed_spec(req.obj) {
         // Creation fails because the spec of the provided object is not well formed
         Option::Some(APIError::BadRequest) // TODO: should the error be BadRequest?
     } else if s.resources.dom().contains(req.obj.set_namespace(req.namespace).object_ref()) {
         // Creation fails because the object already exists
         Option::Some(APIError::ObjectAlreadyExists)
+    } else if req.obj.kind == Kind::CustomResourceKind && !K::rule(req.obj) {
```

Comment:
I would suggest let's unmarshal the object here (K::from_dynamic) and pass the unmarshalled object to rule (and transition rule).